### ISSUE TITLE
Bug: Accessibility issues in user's settings page - `user/settings`

### ISSUE THREAD

**Issue by 5hraddha** [open-sauced/app#2045]
### Describe the bug

There are a few accessibility issues in the user's settings page - `user/settings` that we can improve:

1. **Problem**: Missing form labels for:

      - **Bio** text area,
      -  **Time zone** dropdown,
      -  checkboxes on the page. 
    
    **Possible Solution**: Bind the `for` attribute in the labels to the id of the element this label is associated with. 

2. **Problem**: The first level heading is missing.
    **Possible Solution**: Either keep _Public profile_ as the first level heading or keep _User Settings_ as the first level heading and keep it visually hidden.

If the team is fine with the proposed improvements, I could implement it. Appreciate your suggestions or comments in this regard. Thank you! 

### Steps to reproduce

1. Go to `https://app.opensauced.pizza/user/settings` after logging in.
2. Run an accessibility check. I'm using WAVE (Web Accessibility Evaluation Tool).

### Browsers

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by nickytonline** [comment]
Thanks for opening the issue @5hraddha! I'm aware of our heading issues. It's an app wide thing that needs to be addressed.

For the Bio, the label isn't linked as you mentioned. You can use the `for` attribute (`htmlFor` in JSX), or you can wrap the textarea with the label like this.

```html
<label class="flex flex-col w-full text-sm font-normal text-light-slate-11">
    Bio
    <textarea rows="4" placeholder="Tell us about yourself." class="px-3 py-2 rounded-lg bg-light-slate-4 disabled:cursor-not-allowed " name="bio">Senior Software Engineer at OpenSauced</textarea>
</label>
```

This is what our `TextInput` component does. You can do the same thing for the timezone.

**Comment by 5hraddha** [comment]
Yeah, I saw that. Thanks, Nick! I'll .take the issue then.
